pkg/cgroup/cgroup.go

```diff
@@ -13,36 +13,167 @@ import (
 	"github.com/aquasecurity/tracee/pkg/mount"
 )
 
-const procCgroups = "/proc/cgroups"
-const CgroupV1Controller = "cpuset"
-const releseAgentPath = "/sys/fs/cgroup/cpuset/release_agent"
-const CgroupControllers = "/sys/fs/cgroup/cgroup.controllers"
-const CgroupV1FsType = "cgroup"
-const CgroupV2FsType = "cgroup2"
+type Cgroup interface {
+	Initialize() error
+	Destroy() error
+	GetMountPoint() string
+	GetDefaultHierarchy() int
+	IsV1() bool
+}
 
-// GetCgroupV1HierarchyId returns the hierarchyId of cgroupsv1
-func GetCgroupV1HierarchyId() (int, error) {
-	file, err := os.Open(procCgroups)
+// cgroupv1
+
+func InitCgroupV1() Cgroup {
+	return &cgroupV1{}
+}
+
+type cgroupV1 struct {
+	mountPath string
+	hierarchy int
+}
+
+func (c *cgroupV1) Initialize() error {
+
+	path, err := os.MkdirTemp(TmpCgroup, TmpCgroupV1Path)
 	if err != nil {
-		return -1, fmt.Errorf("could not open cgroups file %s: %w", procCgroups, err)
+		return fmt.Errorf("cgroup: failed to create temporary dir: %s", err)
+	}
+	mountPath, err := filepath.Abs(path)
+	if err != nil {
+		return fmt.Errorf("cgroup: failed to get absolute path: %s", path)
+	}
+	mp, _ := mount.IsMountpointOf(mountPath, CgroupV1FsType)
+	if mp {
```

Comment:
If we're already mounted I think we can just return successful no?

Reply:
It should be mounted only once per temporary tmp directory (with no recurrent callings IMO). Lets see the final version and we can change this if needed.